# National Parks Service #

You will perform some data analysis on the conservation statuses of these species and investigate if there are any patterns or themes to the types of species that become endangered. During this project, you will analyze, clean up, and plot data as well as pose questions and seek to answer them in a meaningful way.

> All imports will appear in the following section for ease of adding to at a later date:

In [1]:
import pandas as pd
import numpy as np

first of all I will read the csv files into a pandas DataFrame:

In [2]:
observations = pd.read_csv('observations.csv')
species_info = pd.read_csv('species_info.csv')

Then view the first 5 rows of each DataFrame

In [3]:
print(observations.head())


            scientific_name                            park_name  observations
0        Vicia benghalensis  Great Smoky Mountains National Park            68
1            Neovison vison  Great Smoky Mountains National Park            77
2         Prunus subcordata               Yosemite National Park           138
3      Abutilon theophrasti                  Bryce National Park            84
4  Githopsis specularioides  Great Smoky Mountains National Park            85


In [4]:
print(species_info.head())

  category                scientific_name  \
0   Mammal  Clethrionomys gapperi gapperi   
1   Mammal                      Bos bison   
2   Mammal                     Bos taurus   
3   Mammal                     Ovis aries   
4   Mammal                 Cervus elaphus   

                                        common_names conservation_status  
0                           Gapper's Red-Backed Vole                 NaN  
1                              American Bison, Bison                 NaN  
2  Aurochs, Aurochs, Domestic Cattle (Feral), Dom...                 NaN  
3  Domestic Sheep, Mouflon, Red Sheep, Sheep (Feral)                 NaN  
4                                      Wapiti Or Elk                 NaN  


Further exploration with info(), columns, nunique, and describe()

In [5]:
print(observations.columns)
print(observations.info())

Index(['scientific_name', 'park_name', 'observations'], dtype='object')
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23296 entries, 0 to 23295
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   scientific_name  23296 non-null  object
 1   park_name        23296 non-null  object
 2   observations     23296 non-null  int64 
dtypes: int64(1), object(2)
memory usage: 546.1+ KB
None


the observations set appears to be clean with 23295 entries. I suspect that some of the scientific names will compare across both data sets. I will explore this at aa later date and maybe look to combine some of the data into a new dataset.

In [6]:
print(species_info.columns)
print(species_info.info())

Index(['category', 'scientific_name', 'common_names', 'conservation_status'], dtype='object')
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5824 entries, 0 to 5823
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   category             5824 non-null   object
 1   scientific_name      5824 non-null   object
 2   common_names         5824 non-null   object
 3   conservation_status  191 non-null    object
dtypes: object(4)
memory usage: 182.1+ KB
None


This data set has 5824 rows. conservation status appears to have a lot of null values: I will need to check the percentage of this column being NAN to see if it meets the 60% missing criteria to delete this column entirely. this would mean that I would not be able to look specifically at the conversation status of all species, I may be restricted at looking at a subset of the data.

Let's look at how many unique scientific names we have in each data set:

In [7]:
print(f'observations: {observations.scientific_name.nunique()}')
print(f'species_info: {species_info.scientific_name.nunique()}')


observations: 5541
species_info: 5541


This looks promising there is exactly the same number of unique scientific names in the data set. I just need to remember how to compare these across two data sets to see if they have the same unique scientific names.

In [12]:
# copy the unique values into a variable for later use:
unique_scientific_observations = observations.scientific_name.unique()
unique_scientific_species = species_info.scientific_name.unique()

In [25]:
unique_scientific_observations.sort
unique_scientific_species.sort

print(unique_scientific_observations)
print(unique_scientific_species)

['Abies bifolia' 'Abies concolor' 'Abies fraseri' ...
 'Zonotrichia querula' 'Zygodon viridissimus'
 'Zygodon viridissimus var. rupestris']
['Abies bifolia' 'Abies concolor' 'Abies fraseri' ...
 'Zonotrichia querula' 'Zygodon viridissimus'
 'Zygodon viridissimus var. rupestris']


From this observation it does not appear that these 2 lists may have the same unique values for 'scientific_name', if this is the case I may be able to clean some of the missing data.